float64 target_pos_threshold
int32 target_pos
float64 current_pos